### ISSUE TITLE
Add `@rspack/core` to the peer dependencies of html-webpack-plugin and react-refresh-webpack-plugin

### ISSUE THREAD

**Issue by chenjiahan** [web-infra-dev/rspack#4537]
### What problem does this feature solve?

Can we add `@rspack/core` to the peer dependencies of html-webpack-plugin and react-refresh-webpack-plugin (and more popular webpack plugins), then make `webpack` as an optional dependency?

- https://github.com/jantimon/html-webpack-plugin/blob/main/package.json#L58-L60
- https://github.com/pmmmwh/react-refresh-webpack-plugin/blob/main/package.json#L114-L123

The benifits:

- Users do not need to install webpack as a dependency.
- No peer dependency missing warning.

### What does the proposed API of configuration look like?

```js
{
  "peerDependencies": {
    "@rspack/core": ">=0.3.0",
    "webpack": ">=4.43.0 <6.0.0",
  },
  "peerDependenciesMeta": {
    "webpack": {
      "optional": true
    },
   "@rspack/core": {
      "optional": true
    }
  },
}
```

**Comment by hardfist** [comment]
yes, I already did this to https://github.com/webpack/webpack-dev-server/pull/4778

**Comment by chenjiahan** [comment]
Update: using forked html-rspack-plugin instead of html-webpack-plugin

https://github.com/rspack-contrib/html-rspack-plugin
